### PR TITLE
added Menu and its test

### PR DESCRIPTION
**Pull request by hendraanggrian**
Following convention of `ViewGroup.kt`, `Menu.kt` does something similar to `MenuItem` as `ViewGroup` does to `View.`

### CODE REVIEW

**Comment by hendraanggrian** [comment]
I signed it!

File: api/0.1.txt [hunk 1]
```diff
@@ -639,6 +639,17 @@ package androidx.util {
 
 package androidx.view {
 
+  public final class MenuKt {
```
Comment (JakeWharton) [inline api/0.1.txt]: Do not change this file.

File: src/main/java/androidx/view/Menu.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+/**
+ * Returns the menu at `index`.
+ *
+ * @throws IndexOutOfBoundsException if index is less than 0 or greater than or equal to the count.
+ */
+inline operator fun Menu.get(index: Int): MenuItem = getItem(index)
+
+/** Returns `true` if `item` is found in this menu. */
+operator fun Menu.contains(item: MenuItem): Boolean {
+    return (0 until size()).any { getItem(it) == item }
```
Comment (JakeWharton) [inline src/main/java/androidx/view/Menu.kt]: Use a normal loop here. This creates a Range and two Iterators which are unnecessary.

File: src/androidTest/java/androidx/view/MenuTest.kt [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import android.support.test.InstrumentationRegistry
+import android.view.MenuItem
+import android.widget.Toolbar
+import androidx.assertThrows
+import androidx.fail
+import com.google.common.truth.Truth.assertThat
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertFalse
+import junit.framework.Assert.assertNotNull
+import junit.framework.Assert.assertTrue
```
Comment (JakeWharton) [inline src/androidTest/java/androidx/view/MenuTest.kt]: Replace all these with `org.junit` imports.

File: src/androidTest/java/androidx/view/MenuTest.kt [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import org.junit.Assert.assertSame
+import org.junit.Test
+
+class MenuTest {
+    private val menu = Toolbar(InstrumentationRegistry.getContext()).menu
+
+    @Test fun get() {
+        val item = menu.add("")
+        assertNotNull(menu[0])
+        assertEquals(item, menu[0])
```
Comment (JakeWharton) [inline src/androidTest/java/androidx/view/MenuTest.kt]: Both of these can be captured in a single `assertSame` call

File: src/androidTest/java/androidx/view/MenuTest.kt [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    @Test fun isEmpty() {
+        assertTrue(menu.isEmpty())
+        menu.add("")
+        assertFalse(menu.isEmpty())
+    }
+
+    @Test fun isNotEmpty() {
+        menu.add("")
```
Comment (JakeWharton) [inline src/androidTest/java/androidx/view/MenuTest.kt]: assertFalse before this call for parity with isEmpty test

File: src/androidTest/java/androidx/view/MenuTest.kt [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+        val items = mutableListOf<MenuItem>()
+        menu.forEachIndexed { index, item ->
+            assertEquals(index, items.size)
+            items += item
+        }
+        assertThat(items).containsExactly(item1, item2)
+    }
+
+    @Test fun iterator() {
```
Comment (JakeWharton) [inline src/androidTest/java/androidx/view/MenuTest.kt]: Nothing tests remove behavior of the iterator

Comment (hendraanggrian) [reply src/androidTest/java/androidx/view/MenuTest.kt]: I'm sorry I don't quite get this one, everything else is fixed as suggested.

File: src/main/java/androidx/view/Menu.kt [hunk 7]
```diff
@@ -30,7 +30,12 @@ inline operator fun Menu.get(index: Int): MenuItem = getItem(index)
 
 /** Returns `true` if `item` is found in this menu. */
 operator fun Menu.contains(item: MenuItem): Boolean {
-    return (0 until size()).any { getItem(it) == item }
+    for (index in 0 until size()) {
+        if (getItem(index) == item) {
+            return false
```
Comment (rom4ek) [inline src/main/java/androidx/view/Menu.kt]: Should be `return true`

Comment (hendraanggrian) [reply src/main/java/androidx/view/Menu.kt]: my bad, just fixed it

File: api/0.1.txt [hunk 8]
```diff
@@ -670,4 +670,3 @@ package androidx.view {
   }
 
 }
-
```
Comment (JakeWharton) [inline api/0.1.txt]: revert changes to this file please

File: src/androidTest/java/androidx/view/MenuTest.kt [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+
+    @Test fun get() {
+        val item = menu.add("")
+        assertSame(item, menu[0])
+    }
+
+    @Test fun contains() {
+        val item1 = menu.add("")
+        assertTrue(item1 in menu)
+        assertFalse(item1 !in menu)
```
Comment (JakeWharton) [inline src/androidTest/java/androidx/view/MenuTest.kt]: This second assertion is testing kotlin and can be removed

File: src/androidTest/java/androidx/view/MenuTest.kt [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+
+        val item2 = menu.add("")
+        assertTrue(item2 in menu)
+        assertFalse(item2 !in menu)
```
Comment (JakeWharton) [inline src/androidTest/java/androidx/view/MenuTest.kt]: ditto

File: src/androidTest/java/androidx/view/MenuTest.kt [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+        assertFalse(item1 in menu)
+        assertEquals(1, menu.size())
+
+        assertSame(item2, iterator.next())
+        iterator.remove()
+        assertFalse(item2 in menu)
+        assertEquals(0, menu.size())
+    }
+
+    @Test fun iteratorForEach() {
```
Comment (JakeWharton) [inline src/androidTest/java/androidx/view/MenuTest.kt]: This is testing the same thing as `iterator()` but with a syntax. The test can be removed.

Comment (hendraanggrian) [reply src/androidTest/java/androidx/view/MenuTest.kt]: Removed.

File: api/0.1.txt (hunk 8 above)
Comment (hendraanggrian) [reply api/0.1.txt]: Reverted.

File: src/androidTest/java/androidx/view/MenuTest.kt (hunk 9 above)
Comment (hendraanggrian) [reply src/androidTest/java/androidx/view/MenuTest.kt]: Removed. It was originally there because I followed `ViewGroupTest.contains()`.

**Comment by JakeWharton** [comment]
Squashed, fixed up a few things, and merged. Thanks!

**Pull request closed without merging**